x-pack/legacy/plugins/siem/public/containers/matrix_histogram/index.ts

```diff
@@ -109,16 +94,7 @@ export const useQuery = <Hit, Aggs, TCache = object>({
       isSubscribed = false;
       abortCtrl.abort();
     };
-  }, [
-    defaultIndex,
-    query,
-    filterQuery,
-    isInspected,
-    isDnsHistogram,
-    stackByField,
-    startDate,
-    endDate,
-  ]);
+  }, [defaultIndex, filterQuery, isInspected, stackByField, startDate, endDate, data]);
```

Comment:
Consider moving the the `const matrixHistogramVariables` definition inside the `useEffect` block, i.e.:

```
  useEffect(() => {
    const matrixHistogramVariables: GetMatrixHistogramQuery.Variables = {
      filterQuery: createFilter(filterQuery),
      sourceId: 'default',
      timerange: {
        interval: '12h',
        from: startDate!,
        to: endDate!,
      },
      defaultIndex,
      inspect: isInspected,
      stackByField,
      histogramType,
    };
```

, because `matrixHistogramVariables` is only referenced within `useEffect`, and moving it there will make it a bit easier to understand why some of the dependencies are specified.

For example, after moving `matrixHistogramVariables` within the block, it's now clear why `endDate` was listed as a dependency.